src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugModule.cs

```diff
@@ -56,7 +61,8 @@ public ResultWrapper<int> debug_deleteChainSlice(in long startNumber)
 
         public ResultWrapper<GethLikeTxTrace> debug_traceTransaction(Keccak transactionHash, GethTraceOptions options = null)
         {
-            GethLikeTxTrace transactionTrace = _debugBridge.GetTransactionTrace(transactionHash, options);
+            CancellationToken cancellationToken = new CancellationTokenSource(_traceTimeout).Token;
```

Comment:
I think we may need to dispose the source at the end. Let me check after merging - I will fix it if needed.